nmostesting/suites/IS0404Test.py

```diff
@@ -0,0 +1,314 @@
+        """
+        # The NCuT shall discover and dynamically update all the Senders that are registered
+        # in the Registry.
+        # * Use the NCuT to browse and take note of the Senders that are available.
+        # * After the 'Next' button has been clicked one of those Senders will be put 'offline'.
+        # * The Sender which was put 'offline' will then come back online at a random moment within
+        #   the following 60 seconds.
+        # * As soon as the NCuT detects the Sender has come back online the user must press the
+        #   'Next' button on the Testing Façade.
+        # * The button must be pressed within 30 seconds of the Sender being put back 'online'.
+        #   This includes any latency between the Sender being put 'online' and the NCuT updating.
+
+        try:
+            # Check senders
+
+            question = """\
+                       The NCuT should be able to discover and dynamically update all the Senders \
+                       that are registered in the Registry.
+
+                       Use the NCuT to browse and take note of the Senders that are available.
+
+                       After the 'Next' button has been clicked one of those senders will be put 'offline'.
+                       """
+            possible_answers = []
+
+            self._invoke_testing_facade(question, possible_answers, test_type="action")
+
+            # Take one of the senders offline
+            possible_answers = [{'answer_id': 'answer_'+str(i), 'label': s['label'],
+                                 'description': s['description'], 'id': s['id'], 'answer_str': s['answer_str']}
+                                for i, s in enumerate(self.senders) if s['registered']]
+            answer_indices = [index for index, s in enumerate(self.senders) if s['registered']]
+            offline_sender_index = random.choice(answer_indices)
+            expected_answer = 'answer_' + str(offline_sender_index)
+
+            self._delete_sender(test, self.senders[offline_sender_index])
+
+            # Set the offline sender to registered false for future tests
+            self.senders[offline_sender_index]['registered'] = False
+
+            # Recheck senders
+            question = 'Please refresh your NCuT and select the sender which has been put \'offline\'.'
+
+            actual_answer = self._invoke_testing_facade(
+                question, possible_answers, test_type="single_choice", multipart_test=1)['answer_response']
+
+            if actual_answer != expected_answer:
+                return test.FAIL('Offline/online sender not handled: Incorrect sender identified')
+
+            max_time_until_online = 60
+            max_time_to_answer = 30
+
+            question = """\
+                       The sender which was put 'offline' will come back online at a random moment \
+                       within the next \
+                       """\
+                       + str(max_time_until_online) + \
```

Comment:
https://github.com/AMWA-TV/nmos-testing/pull/616#pullrequestreview-835347438

Reply:
Formatting of questions now generally improved though use of multi-line strings and `dedent`